packages/@aws-cdk/aws-apprunner/lib/service.ts

```diff
@@ -806,6 +814,10 @@ export class Service extends cdk.Resource {
           vpcConnectorArn: this.props.vpcConnector?.vpcConnectorArn,
         },
       },
+      observabilityConfiguration: {
+        observabilityConfigurationArn: this.props.observabilityConfiguration?.observabilityConfigurationArn,
+        observabilityEnabled: !!this.props.observabilityConfiguration,
+      },
```

Comment:
If `observabilityConfiguration` isn't defined, just don't pass it here. We don't need to edit templates if no changes are actually made. This will significantly cut down on the changes showing up in tests where you aren't actually creating this construct.